src/multidoc-manager.js

```diff
@@ -358,6 +359,24 @@ export class MultidocManager {
               // Ignore.
             } else if (name == 'viewport') {
               // Ignore.
+            } else if (name == 'amp-script-src') {
+              // Import amp-script hashes, concatenating content attributes into
+              // a single <meta name="amp-script-src"> tag in the event of
+              // multiple ampdocs.
+              let el = this.win.document.head.querySelector(
+                `meta[name="${escapeCssSelectorIdent(name)}"]`
+              );
+              if (!el) {
+                el = this.win.document.createElement('meta');
+                el.setAttribute('name', name);
+                this.win.document.head.appendChild(el);
+              }
+              const content = (
+                (el.getAttribute('content') || '') +
+                ' ' +
+                (n.getAttribute('content') || '')
+              ).trim();
+              el.setAttribute('content', content);
```

Comment:
I'd prefer to keep the script hashes separate between documents (to avoid allowing a document to enable scripts on another). Could we instead just import these nodes and append them to the shadow root? 

We do something similar for JSON scripts: https://github.com/ampproject/amphtml/pull/26535/files#diff-0a54674fbde60777f171c0027199f578L452

Then we can revert the amp-script.js changes.

/cc @dvoytenko

Reply:
We could. I was avoiding insertion of meta tags into `ShadowRoot` since they should only appear under the `<head>` element. Do you have a preference for one of the following?
1. allow `<meta>` in `ShadowRoot` despite invalid HTML
2. import script hashes into a dedicated, invisible element in `ShadowRoot`

Reply:
Fair point. We don't really need this to exist in the shadow DOM, so maybe we can add a metadata abstraction to the `AmpDoc` class instead.

I.e. similar to the existing `AmpDoc.getParam()`, add `AmpDoc.getMeta(name)` which returns the "content" attribute value for the meta tag with the corresponding name. `AmpDocShadow` would populate this in `mergeShadowHead_`, and `AmpDocSingle` would query its head node.